src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2145,12 +2165,33 @@ private ushort getRelocTypeHint(IntPtr _this, void* target)
             return UInt16.MaxValue;
         }
 
-        private void getModuleNativeEntryPointRange(IntPtr _this, ref void* pStart, ref void* pEnd)
+        private void getModuleNativeEntryPointRange(ref void* pStart, ref void* pEnd)
         { throw new NotImplementedException("getModuleNativeEntryPointRange"); }
 
-        private uint getExpectedTargetArchitecture(IntPtr _this)
+        private uint getExpectedTargetArchitecture()
         {
             return 0x8664; // AMD64
         }
+
+        [UnmanagedFunctionPointerAttribute(CallingConvention.ThisCall)]
+        delegate void _getLocationOfThisType_wrapper_Windows(IntPtr _this, CORINFO_LOOKUP_KIND* result, out IntPtr exception, CORINFO_METHOD_STRUCT_* context);
+
+        void FixupUnmanagedInstance()
+        {
+            if (IsWindows())
+            {
+                IntPtr * vtable = *(IntPtr**)_comp;
+                for (int i = 0; i < _keepalive.Length; i++)
+                {
```

Comment:
We should add concept of manually written C++  wrapper. It should be used for getLocationOfThisType and FilterException:

- getLocationOfThisType: the managed side should always get the return value as [out] argument. The workaround on the managed side should go away, and the  special cases on the unmanaged side will be centralized.

- FilterException should just unconditionally return EXECUTE_HANDLER on the C++ side without calling managed code. (It is a reliability problem to write Win32 unmanaged exception filters in managed code.)